compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1739,7 +1739,8 @@ private boolean isValidInferredArray(BLangNode node) {
             case VARIABLE:
                 BLangSimpleVariable varNode = (BLangSimpleVariable) node;
                 BLangExpression expr = varNode.expr;
-                return expr != null && expr.getKind() == NodeKind.LIST_CONSTRUCTOR_EXPR &&
+                return expr != null && (expr.getKind() == NodeKind.LIST_CONSTRUCTOR_EXPR ||
+                        expr.getKind() == LITERAL) &&
```

Comment:
For below expr kind will be group-expr. Hence, the logic will break
```ballerina
 byte[*] _ = (base16 `aa bb`);
```

Reply:
Updated 97e8d52b62113bda3abb212078e222f001bea4f3